lib/archethic/self_repair/scheduler.ex

```diff
@@ -92,33 +94,56 @@ defmodule Archethic.SelfRepair.Scheduler do
       # Loading transactions can take a lot of time to be achieve and can overpass an epoch.
       # So to avoid missing a beacon summary epoch, we save the starting date and update the last sync date with it
       # at the end of loading (in case there is a crash during self repair)
-      start_date = DateTime.utc_now()
-      :ok = Sync.load_missed_transactions(last_sync_date)
-      {:ok, start_date}
+      Sync.load_missed_transactions(last_sync_date)
     end)
 
+    {:noreply, state, :hibernate}
+  end
+
+  def handle_info({ref, :ok}, state = %{interval: interval}) do
+    # If the node is still unavailable after self repair, we send the postpone the
+    # end of node sync message
     timer = schedule_sync(interval)
     Logger.info("Self-Repair will be started in #{Utils.remaining_seconds_from_timer(timer)}")
 
     new_state =
       state
       |> Map.put(:timer, timer)
+      |> Map.delete(:retry_count)
 
-    {:noreply, new_state, :hibernate}
-  end
+    if !Archethic.up?() do
+      :persistent_term.put(:archethic_up, :up)
+      PubSub.notify_node_status(:node_up)
+    end
 
-  def handle_info({ref, {:ok, date}}, state) do
-    # If the node is still unavailable after self repair, we send the postpone the
-    # end of node sync message
     if !P2P.available_node?(), do: BootstrapSync.publish_end_of_sync()
-    update_last_sync_date(date)
     Process.demonitor(ref, [:flush])
-    {:noreply, state}
+    {:noreply, new_state}
   end
 
-  def handle_info({:DOWN, _ref, _, _, reason}, state) do
+  def handle_info({:DOWN, _ref, _, _, reason}, state = %{interval: interval}) do
```

Comment:
The DOWN message is received, even in the case of success, only when the task process will be terminated.
So, this is not a failure and the retry is irrelevant in that case.

Reply:
Instead, you might use a try catch in the `Task.async` and return `error` in the catch.
Hence, you could pattern match on the `ok/error` tuple to determine the success or the retry.

Reply:
The DOWN event is not always receive because in the handle info of a success task ({ref, response}) we use `Process.demonitor(ref [:flush])` In that case the message DOWN will not be receive by the scheduler. Otherwise we should have a failed self-repair error log after all self repair even if it is successfull

Reply:
ok